### PR TITLE
setup indexing

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: ingest.py [hunk 1]
```diff
@@ -82,17 +88,38 @@ def ingest_docs():
     # with open('docs_transformed.pkl', 'rb') as f:
     #     docs_transformed = pickle.load(f)
     
-    client = weaviate.Client(url=WEAVIATE_URL, auth_client_secret=weaviate.AuthApiKey(api_key=WEAVIATE_API_KEY))
-    client.schema.delete_class("LangChain_newest_idx") # delete the class if it already exists
+    client = weaviate.Client(
+        url=WEAVIATE_URL,
+        auth_client_secret=weaviate.AuthApiKey(api_key=WEAVIATE_API_KEY)
+    )
+    embedding = OpenAIEmbeddings(chunk_size=200)  # rate limit
```
Comment (eyurtsev) [inline ingest.py:95]: why does this say rate limit?

File: ingest.py [hunk 2]
```diff
@@ -82,17 +88,38 @@ def ingest_docs():
...
+    record_manager = SQLRecordManager(
+        RECORD_MANAGER_NAMESPACE,
+        db_url=RECORD_MANAGER_DB_URL
+    )
+    record_manager.create_schema()
+    index(
+        docs_transformed,
+        record_manager,
+        vectorstore,
+        delete_mode="full",
```
Comment (eyurtsev) [inline ingest.py:114]: If we're using `full` should we cache the embeddings?

File: ingest.py [hunk 3]
```diff
@@ -82,17 +88,38 @@ def ingest_docs():
...
+        by_text=False,
+        attributes=["source"]
+    )
 
-    embeddings = OpenAIEmbeddings(chunk_size=200) # rate limit
+    record_manager = SQLRecordManager(
+        RECORD_MANAGER_NAMESPACE,
+        db_url=RECORD_MANAGER_DB_URL
+    )
+    record_manager.create_schema()
```
Comment (eyurtsev) [inline ingest.py:109]: I haven't double checked what happens if this is called twice. Semantics we want here is create if needed, but potentially may raise an exception on second create

File: requirements.txt [hunk 4]
```diff
@@ -13,4 +13,5 @@ unstructured
 libmagic
 tiktoken
 weaviate-client
-anthropic
\ No newline at end of file
+anthropic
```
Comment (eyurtsev) [inline requirements.txt:16]: we should really be using pyproject toml everywhere by default

**Review by eyurtsev — APPROVED** [review]

File: ingest.py (hunk 1 above)
Comment (mcantillon21) [reply ingest.py:95]: when upserting into weaviate, we ran into openai rate limits with default chunk sizes, so declaring it to be 200 helped deal with that

File: ingest.py (hunk 2 above)
Comment (baskaryan) [reply ingest.py:114]: are embeddings recomputed for existing docs? thought vectorstore.add_documents only gets called after record manager checks for changes

**Review by mcantillon21 — APPROVED** [review]

File: ingest.py (hunk 2 above)
Comment (eyurtsev) [reply ingest.py:114]: yeah apologies, we should be good to go i confused myself

**Pull request merged**